samples/net/sockets/http_server/src/main.c

```diff
@@ -37,59 +44,139 @@ struct http_resource_detail_static index_html_gz_resource_detail = {
 	.static_data_len = sizeof(index_html_gz),
 };
 
-HTTP_RESOURCE_DEFINE(index_html_gz_resource, test_http_service, "/",
-		     &index_html_gz_resource_detail);
+static struct http_resource_detail_static main_js_gz_resource_detail = {
+	.common = {
+			.type = HTTP_RESOURCE_TYPE_STATIC,
+			.bitmask_of_supported_http_methods = BIT(HTTP_GET),
+			.content_encoding = "gzip",
+			.content_type = "text/javascript",
+		},
+	.static_data = main_js_gz,
+	.static_data_len = sizeof(main_js_gz),
+};
 
-static uint8_t recv_buffer[1024];
+static int uptime_handler(struct http_client_ctx *client, enum http_data_status status,
+			  uint8_t *buffer, size_t len, void *user_data)
+{
+	static bool response_sent;
+
+	LOG_DBG("Uptime handler status %d, response_sent %d", status, response_sent);
 
-static int dyn_handler(struct http_client_ctx *client, enum http_data_status status,
+	switch (status) {
+	case HTTP_SERVER_DATA_ABORTED:
+	{
```

Comment:
Minor style nit, please place the starting `{` to the previous line here and the lines below